src/main/java/duke/duke/info/task/Task.java

```diff
@@ -0,0 +1,134 @@
+package duke.info.task;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+import java.time.format.DateTimeFormatter;
+
+public abstract class Task {
+
+    private final String action;
+    private boolean isComplete;
+    private final String type;
+    private String dateString;
+    private LocalDate date;
+
+    /**
+     * Constructor for an abstract Task uses the specified action, type and
+     * isComplete boolean. The dateString and date is instantiated as null
+     * @param action - the action of the task
+     * @param type - the type of the task
+     * @param isComplete - whether the task is complete
+     */
+    Task(String action, String type, boolean isComplete) {
+        this.action = action;
+        this.type = type;
+        this.isComplete = isComplete; // task added is not complete by default
+        this.dateString = null;
+        this.date = null;
+    }
+
+    /**
+     * Constructor for an abstract Task with a specified date that uses the
+     * specified action, type, isComplete boolean and dateString. If the dateString
+     * can be parsed as a LocalDate, it will be saved as a LocalDate object. Otherwise,
+     * it will be saved as a raw dateString.
+     * @param action - the action of the task
+     * @param type - the type of the task
+     * @param isComplete - whether the task is complete
+     * @param dateString - the string representation of the date of the task
+     */
+    Task(String action, String type, boolean isComplete, String dateString) {
+        this.action = action;
+        this.type = type;
+        this.isComplete = isComplete;
+        try {
+            this.date = LocalDate.parse(dateString);
+            this.dateString = null;
+        } catch (DateTimeParseException e) {
+            this.date = null;
+            this.dateString = dateString;
+        }
+    }
+
+    /**
+     * Marks the task as complete by setting isComplete to true
+     */
+    void complete() {
```

Comment:
Should this better rephrased?
```suggestion
    void markTaskComplete() {
```